src/fancy/Grid.hx

```diff
@@ -190,32 +203,35 @@ class Grid {
     };
   }
 
-  function assignHSize(f: Int -> Float): Int -> Float {
-    if(null != f) return f;
+  function assignHSize(): Int -> Float {
     var cache = new IntCache();
     caches.push(cache);
-    return function(col) {
+
+    return function(col: Int) {
       if(cache.exists(col))
         return cache.get(col);
+
       var v = 0.0;
-      // test headers and first content cell
       var els: Array<Element> = [], el;
-      for(i in 0...(fixedTop + 1).max(2)) {
-        el = renderAt(i, col);
-        els.push(el);
-        view.append(el);
-      }
-      // test last content cell and footer
-      for(i in rows - fixedBottom - 1...rows) {
-        el = renderAt(i, col);
-        els.push(el);
-        view.append(el);
+
+      for (i in 0...rows) {
+        // render each row that is in the fixed top, or the first row after
+        // fixed top, or the last row before fixed bottom, or in fixed bottom
+        if (i < fixedTop + 1 || i > rows - fixedBottom - 1) {
```

Comment:
you killed my optimization ... with 1.000.000 rows this check will be performed that many times. It is certainly a fast check so maybe it is fine this way.